src/background/service-worker-init.ts

```diff
@@ -52,6 +52,7 @@ async function initialize(): Promise<void> {
     const promiseFactory = createDefaultPromiseFactory();
     const browserEventProvider = new BrowserEventProvider();
     const browserEventManager = new BrowserEventManager(promiseFactory, logger, true);
+    // Create browser adapter before any async calls to ensure that browser listeners are registered first
```

Comment:
This comment is explaining what is happening, but not *why* it's happening, which I think is generally much more valuable information for future maintainers to have available. A reader can already see that the adapter is being created before any async calls, but a reader without context won't know that the reason why this is important is that if the service worker does not register all of its browser listeners *synchronously* during worker initialization, the browser may tear down the worker as being "done" before we ever register any listeners.

```suggestion
    // It is important that the browser adapter gets initialized *before* any "await" statement.
    //
    // If a service worker does not register all of its browser listeners *synchronously* during worker initialization,
    // the browser may decide that the worker is "done" as soon as the synchronous part of initialization finishes
    // and tear down the worker before we tell it which events to wake us back up for.
```